File: app/Telegram/Guards.php
<?php

namespace App\Telegram;

use Telegram\Bot\Laravel\Facades\Telegram;

class Guards
{
    public static function isAdmin(int $userId): bool
    {
        $raw = config('telegram-bot.admin_ids', '');
        if (empty($raw)) {
            return false;
        }
        $ids = array_map('intval', array_filter(explode(',', $raw)));
        return in_array($userId, $ids, true);
    }

    public static function requireAdmin(int $userId, string $callbackQueryId): bool
    {
        if (self::isAdmin($userId)) {
            return true;
        }
        Telegram::answerCallbackQuery([
            'callback_query_id' => $callbackQueryId,
            'text'              => 'Admin access required.',
            'show_alert'        => true,
        ]);
        return false;
    }
}

File: handlers/guards.py
from telegram import Update
from config import ADMIN_IDS


async def _require_admin(update: Update) -> bool:
    """Return True if caller is admin; otherwise respond and return False."""
    if update.effective_user.id in ADMIN_IDS:
        return True
    if update.callback_query:
        await update.callback_query.answer("⛔ Admin access required.", show_alert=True)
    else:
        await update.effective_message.reply_text("⛔ You don't have admin access.")
    return False
